components/layout_2020/flow/mod.rs

```diff
@@ -1621,9 +1617,11 @@ impl PlacementState {
                     self.current_margin
                         .adjoin_assign(&fragment_block_margins.start);
                 }
+                self.current_block_direction_position += self.current_margin.solve().into();
 
-                fragment.content_rect.start_corner.block +=
-                    self.current_margin.solve() + self.current_block_direction_position;
+                fragment.content_rect.start_corner.block += (self.current_margin.solve() +
+                    self.current_block_direction_position.into())
+                .into();
```

Comment:
Avoid back-and-forth conversions:

```suggestion
                fragment.content_rect.start_corner.block +=
                    Length::from(self.current_margin.solve()) +
                        self.current_block_direction_position;
```